internal/samsahai/activepromotion/promote.go

```diff
@@ -52,9 +52,19 @@ func (c *controller) promoteActiveEnvironment(ctx context.Context, atpComp *s2hv
 		"Result has been collected, promoted successfully")
 	atpComp.Status.SetCondition(s2hv1.ActivePromotionCondActivePromoted, corev1.ConditionTrue,
 		"Active environment has been promoted")
-	atpComp.SetState(s2hv1.ActivePromotionDestroyingPreviousActive,
-		"Destroying the previous active environment")
 
+	if *atpComp.Spec.NoDowntimeGuarantee {
```

Comment:
also this one

it's because, the first time that we upgrade Samsahai, this can be nil if there are some teams are at this state

Reply:
fixed, please help check again